app/code/Magento/QuoteGraphQl/Model/Cart/SetShippingAddressOnCart.php

```diff
@@ -67,6 +67,11 @@ public function __construct(
      */
     public function execute(ContextInterface $context, CartInterface $cart, array $shippingAddresses): void
     {
+        if ($cart->getIsVirtual()) {
```

Comment:
We need to move this check on a higher level to `\Magento\QuoteGraphQl\Model\Resolver\SetShippingAddressesOnCart::resolve`
Also. need to cover case with API-functional tests

Reply:
Done. add try-catch wrapper and api-functional test